src/main/webapp/app/exam/participate/exam-participation.service.ts

```diff
@@ -246,4 +246,42 @@ export class ExamParticipationService {
             });
         }
     }
+
+    public static getSubmissionForExercise(exercise: Exercise) {
+        if (
+            exercise &&
+            exercise.studentParticipations &&
+            exercise.studentParticipations.length > 0 &&
+            exercise.studentParticipations[0].submissions &&
+            exercise.studentParticipations[0].submissions.length > 0
+        ) {
+            return exercise.studentParticipations[0].submissions[0];
+        } else {
+            return undefined;
+        }
+    }
+
+    getExerciseButtonTooltip(
+        exercise: Exercise,
+        submission = ExamParticipationService.getSubmissionForExercise(exercise),
+    ): 'submitted' | 'notSubmitted' | 'synced' | 'notSynced' | 'notSavedOrSubmitted' {
+        // submission does not yet exist for this exercise.
+        // When the participant navigates to the exercise the submissions are created.
+        // Until then show, that the exercise is synced
+        if (!submission) {
+            return 'synced';
+        }
+        if (exercise.type !== ExerciseType.PROGRAMMING) {
+            return submission.isSynced ? 'synced' : 'notSynced';
+        }
+        if (submission.submitted && submission.isSynced) {
+            return 'submitted'; // You have submitted an exercise. You can submit again
+        } else if (submission.submitted && !submission.isSynced) {
+            return 'notSavedOrSubmitted'; // You have unsaved and/or unsubmitted changes
+        } else if (!submission.submitted && submission.isSynced) {
+            return 'notSubmitted'; // starting point
+        } else {
+            return 'notSavedOrSubmitted';
```

Comment:
```suggestion
        if (submission.submitted && submission.isSynced) {
            return 'submitted'; // You have submitted an exercise. You can submit again
        } else if (!submission.submitted && submission.isSynced) {
            return 'notSubmitted'; // starting point
        } else {
            return 'notSavedOrSubmitted';
```
Should also work as we return the same string in two different scenarios

Reply:
good catch!